fuzz/__main__.py

```diff
@@ -13,83 +13,129 @@
-                    for candidate in candidates:
-                        api_validation.validate(candidate)
-        else:
-            # print(" requests.good ")
-            response = Response()
-            # offset:0x0600
-            content = fdp.ConsumeBytes(512)
-            response._content = content
-            status_codes = [0, 200, 400, 401, 403]
-            # maximum 16 items due only 4 bits is used from mock_fuzz_byte
-            assert len(status_codes) <= 0x10
-            response.status_code = status_codes[(0xF & fuzz_byte) % len(status_codes)]
-            with patch("requests.get", return_value=response):
-                with patch("requests.post", return_value=response):
-                    for candidate in candidates:
-                        api_validation.validate(candidate)
+    to_scan = fdp.ConsumeBytes(INPUT_DATA_SIZE)
+    logger.debug(">>>>>>>>%s", to_scan.decode(encoding='ascii', errors="ignore"))
+    # seed file name is sha1 of the content
+    file_name = hashlib.sha1(data).hexdigest()
+    provider = DataContentProvider(to_scan, file_name)
+    candidates = cred_sweeper.data_scan(provider, 1, INPUT_DATA_SIZE)
+
+    if INPUT_DATA_SIZE < len(data):
+        # offset:0x0600
+        fuzz_bytes = fdp.ConsumeBytes(1)
+        behaviour_code = 0xF & fuzz_bytes[0]
+        assert 0 <= behaviour_code <= 15
+        status_code_seed = fuzz_bytes[0] >> 4
+        assert 0 <= status_code_seed <= 15
+        # offset:0x0601
+        content: bytes = fdp.ConsumeBytes(MOCK_RESPONSE_SIZE) if 0 == behaviour_code else b''
+        for candidate in candidates:
+            for validation in candidate.validations:
+                if validation.__class__.__name__ in [  #
+                    GithubTokenValidation.__name__,  #
+                    GoogleApiKeyValidation.__name__,  #
+                    MailChimpKeyValidation.__name__,  #
+                    SquareClientIdValidation.__name__,  #
+                    StripeApiKeyValidation.__name__]:
+                    mock_request(behaviour_code, status_code_seed, content, candidate, requests, requests.get.__name__)
+                elif validation.__class__.__name__ in [  #
+                    SquareAccessTokenValidation.__name__,  #
+                    SlackTokenValidation.__name__]:
+                    mock_request(behaviour_code, status_code_seed, content, candidate, requests, requests.post.__name__)
+                elif validation.__class__.__name__ in [GoogleMultiValidation.__name__]:
+                    mock_flow(behaviour_code, candidate)
 
     cred_sweeper.credential_manager.set_credentials(candidates)
     cred_sweeper.post_processing()
 
 
 def main():
-    # # # Instrument all works with ~20K functions, but it does not lose seeds during reducing
-    if not os.getenv('SKIP_ATHERIS_INSTRUMENT'):
+    # # # Instrument all works with ~26K functions, but it does not lose seeds during reducing
```

Comment:
Three `#` looks very fancy

Reply:
The trick helps to avoid issue with automatic uncomment lines in IDE when real comment line is selected by mistake.